File: scala/scala-impl/testdata/refactoring/rename3/objectAndPrivateClass/before/tests/Foo.java
package tests;

public class Foo {
    public static void main(String[] args) {
        Bar/*caret*/.x();
        Bar$/*caret*/.MODULE$.x();

        Bar/*caret*/ b1 = new /*caret*/Bar();
        b1.baz();

        Bar/*caret*/ b2 = new /*caret*/Bar(0);
        b2.baz();
    }
}

File: scala/scala-impl/testdata/refactoring/rename3/classAndCompanionObjectEndMarker/before/tests/Foo.scala
package tests

class /*caret*/Foo:
  def bar: Int = 42
end Foo

object Foo/*caret*/:
  def apply() = new Foo/*caret*/()
end Foo

class Bar extends Foo/*caret*/:
  val baz = "..."
end Bar

object Test:
  val foo = Foo/*caret*/()
  val bar = new Bar()
end Test
